src/Utilities/Compiler/Options/AbstractCategorizedAnalyzerConfigOptions.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities
+{
+    using static CategorizedAnalyzerConfigOptionsExtensions;
+
+    internal abstract class AbstractCategorizedAnalyzerConfigOptions : ICategorizedAnalyzerConfigOptions
```

Comment:
Most of the code here is existed code from `CategorizedAnalyzerConfigOptions.cs` just extracted to a separate base type.